app/src/main/java/com/humara/nagar/ui/report/ReportFragment.kt

```diff
@@ -23,23 +43,216 @@ class ReportFragment : BaseFragment() {
+        reportViewModel.setLocation(addressLine)
+    }
+
+    private fun checkForLocationPermission() {
+        val permissions = arrayOf(
+            Manifest.permission.ACCESS_FINE_LOCATION,
+            Manifest.permission.ACCESS_COARSE_LOCATION
+        )
+        if (!PermissionsUtils.checkPermissions(requireContext(), permissions)) {
+            PermissionsUtils.requestPermissions(this@ReportFragment, permissions)
+            requestedPermissions.clear()
+            requestedPermissions = permissions.toMutableList()
+        } else getAddress()
+    }
+
+    private fun showPictureDialog() {
+        val pictureDialog = AlertDialog.Builder(requireContext())
+        pictureDialog.setTitle(resources.getString(R.string.selectAction))
+        val pictureDialogItems = arrayOf(
+            resources.getString(R.string.selectFromGallery),
+            resources.getString(R.string.captureFromCamera)
+        )
+        pictureDialog.setItems(
+            pictureDialogItems
+        ) { _, which ->
+            when (which) {
+                0 -> choosePhotoFromGallery()
+                1 -> takePhotoFromCamera()
+            }
+        }
+        val dialog = pictureDialog.create()
+        dialog.show()
+    }
+
+    private fun choosePhotoFromGallery() {
+        val permissions = arrayOf(
+            Manifest.permission.READ_EXTERNAL_STORAGE
+        )
+        if (!PermissionsUtils.checkPermissions(requireContext(), permissions)) {
+            PermissionsUtils.requestPermissions(this@ReportFragment, permissions)
+            requestedPermissions.clear()
+            requestedPermissions = permissions.toMutableList()
+        } else {
+            //Pick multiple pictures (limit max 5)
         }
-        return root
+    }
+
+    private fun takePhotoFromCamera() {
+        val permissions = arrayOf(
+            Manifest.permission.CAMERA
+        )
+        if (!PermissionsUtils.checkPermissions(requireContext(), permissions)) {
+            PermissionsUtils.requestPermissions(this@ReportFragment, permissions)
+            requestedPermissions.clear()
+            requestedPermissions = permissions.toMutableList()
+        } else {
+            //Take picture
+        }
+    }
+
+
+    @Deprecated("Deprecated in Java")
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
+        grantResults: IntArray
+    ) {
+        var allPermissionsGranted = true
+        val deniedPermissions = mutableListOf<String>()
+        for (i in requestedPermissions.indices) {
+            if (grantResults[i] != PackageManager.PERMISSION_GRANTED) {
+                Logger.debugLog("NOT GRANTED ${requestedPermissions[i]}")
+                allPermissionsGranted = false
+                deniedPermissions.add(requestedPermissions[i])
+                break
+            } else {
+                Logger.debugLog("GRANTED ${requestedPermissions[i]}")
+                break
```

Comment:
this break should not be here. If first permission is granted but rest of the permission is denied still allPermissionsGranted would be true

Reply:
Yes, it makes sense, made changes.